test/timeout.js

```diff
@@ -0,0 +1,64 @@
+import test from 'ava';
+import got from '..';
+import {createServer} from './helpers/server';
+
+let s;
+
+test.before('setup', async () => {
+	s = await createServer();
+
+	s.on('/', (req, res) => {
+		res.statusCode = 200;
+		res.end('OK');
+	});
+
+	await s.listen(s.port);
+});
+
+test('timeout option', async t => {
```

Comment:
The rest of the tests tend to take this form:
```js
	const err = await t.throws(got(`${s.url}/`, {
		timeout: 1,
		retries: 0
	}));
	t.is(err.code, 'ETIMEDOUT');
```
I'd be in favor of testing this way because:
* it offloads worrying about if it throws to ava.
* consistent style is helpful in reducing complexity.